From 2ed0351463f6ae6139bedb57fe414d687321b422 Mon Sep 17 00:00:00 2001
From: Richard Zhao <b20223@freescale.com>
Date: Tue, 10 Mar 2009 16:58:32 +0800
Subject: [PATCH] ENGR00109268 unifi_fs: don't reset chip if power is on by default

If power is on by default, sdio device has been initialized, so we don't
reset the chip.

Signed-off-by: Richard Zhao <b20223@freescale.com>
(cherry picked from commit 639db3679a4167d0292f8e67204c348c9ea73f71)
---
 drivers/mmc/card/unifi_fs/fs_lx.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/mmc/card/unifi_fs/fs_lx.c b/drivers/mmc/card/unifi_fs/fs_lx.c
index c2e1d68..15f1722 100644
--- a/drivers/mmc/card/unifi_fs/fs_lx.c
+++ b/drivers/mmc/card/unifi_fs/fs_lx.c
@@ -8,7 +8,7 @@
  *
  */
 /*
- * Copyright 2008 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright 2008-2009 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 /*
  * The code contained herein is licensed under the GNU General Public
@@ -358,8 +358,8 @@ int fs_sdio_register_driver(struct fs_driver *driver)
 {
 	int ret;
 
-	/* Switch us on */
-	fs_unifi_power_on(-1);
+	/* Switch us on, sdio device may exist if power is on by default. */
+	fs_unifi_power_on(available_sdio_dev ? 0 : 1);
 
 	/* Store the context to the device driver to the global */
 	available_driver = driver;
-- 
1.5.4.4

